app/controllers/spree/admin/images_controller_decorator.rb

```diff
@@ -41,10 +41,22 @@ def set_viewable
   private
 
   def create_image(variant, image_attributes)
-    image = Spree::Image.new(permitted_resource_params)
-    image.viewable_type = 'Spree::Variant'
-    image.viewable_id = variant.id
-    variant.images << image
-    variant.save
+
+    attachment = params[:image][:attachment]
+
+    if attachment.nil?
+      image = Spree::Image.new(permitted_resource_params)
+      image.attachment = @image.attachment
+      image.viewable_type = 'Spree::Variant'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.